packages/backend/src/routes/PostRouter.ts

```diff
@@ -233,3 +233,143 @@ async function createPostSubsidy(req, res) {
         post,
     })
 }
+
+async function editPost(req, res) {
+    const id = req.params.id
+    const unirepSocialContract = new ethers.Contract(
+        UNIREP_SOCIAL,
+        UNIREP_SOCIAL_ABI,
+        DEFAULT_ETH_PROVIDER
+    )
+
+    // Parse Inputs
+    const { publicSignals, proof, content } = req.body
+    const epkProof = new EpochKeyProof(
+        publicSignals,
+        formatProofForSnarkjsVerification(proof)
+    )
+    const newHashedContent = ethers.utils.keccak256(
+        ethers.utils.toUtf8Bytes(content)
+    )
+
+    const error = await verifyEpochKeyProof(req.db, epkProof)
+    if (error !== undefined) {
+        res.status(422).json({
+            error,
+        })
+        return
+    }
+
+    const { hashedContent: oldHashedContent, postId } = await req.db.findOne(
+        'Post',
+        {
+            where: {
+                _id: id,
+            },
+        }
+    )
+
+    const calldata = unirepSocialContract.interface.encodeFunctionData('edit', [
+        postId,
+        oldHashedContent,
+        newHashedContent,
+        epkProof.publicSignals,
+        epkProof.proof,
+    ])
+
+    const hash = await TransactionManager.queueTransaction(
+        unirepSocialContract.address,
+        {
+            data: calldata,
+        }
+    )
+
+    await req.db.update('Post', {
+        where: {
+            _id: id,
+            postId,
+            hashedContent: oldHashedContent,
+        },
+        update: {
+            content,
+            hashedContent: newHashedContent,
+        },
+    })
+
+    const post = await req.db.findOne('Post', { where: { _id: id } })
+
+    res.json({
+        error: error,
+        transaction: hash,
+        post,
+    })
+}
+
+async function deletePost(req, res) {
```

Comment:
Same here, on the frontend let's just have it update the post content to `Deleted by user` and remove this route.